.clang-tidy

```
@@ -0,0 +1,12 @@
+# Clang-tidy configuration used on the whole code base
+Checks: >
+  bugprone-unhandled-self-assignment,
+  modernize-deprecated-headers,
+  modernize-redundant-void-arg,
+  modernize-use-bool-literals,
+  modernize-use-nullptr,
+  readability-braces-around-statements,
+  misc-no-recursion,
+  clang-analyzer-*,
+  -clang-analyzer-security.insecureAPI.rand
```

Here is a table of all clang-tidy checks - https://clang.llvm.org/extra/clang-tidy/#using-clang-tidy

I'd say we keep the following for this project:


```suggestion
Checks: >
  bugprone-*,
  modernize-*,
  readability-*,
  clang-analyzer-*,
  cppcoreguidelines-*,
  google-*
```

What do you think?

---

Yes but later we could drop modernize if it keeps proposing things we have not seen in class.